api/v1alpha2/azurecluster_conversion.go

```diff
@@ -63,6 +63,18 @@ func (src *AzureCluster) ConvertTo(dstRaw conversion.Hub) error { // nolint
 			for _, dstSubnet := range dst.Spec.NetworkSpec.Subnets {
 				if dstSubnet != nil && dstSubnet.Name == restoredSubnet.Name {
 					dstSubnet.RouteTable = restoredSubnet.RouteTable
+					dstSubnet.SecurityGroup = restoredSubnet.SecurityGroup
```

Comment:
security group isn't a new field in v1alpha3 so it shouldn't be restored (only IngressRules[i].priority and IngressRules[i].name like you did below)